src/main/groovy/com/jfrog/bintray/gradle/BintrayUploadTask.groovy

```diff
@@ -473,6 +470,21 @@ class BintrayUploadTask extends DefaultTask {
         }
     }
 
+    def assertUploadConfiguration() {
+        def errors = []
+        if(user == null) {
+            errors.add("user is not configured")
+        }
+        if(apiKey == null) {
+            errors.add("key is not configured")
+        }
+        if(!errors.isEmpty()) {
+            errors.join(" and ")
+            def message = "Failing task '${this.project.name}:bintrayUpload' because ${errors.join(" and ")}"
```

Comment:
It is not desirable to duplicate the task name here. Instead of ```${this.project.name}:bintrayUpload``` we can just use ```$path```. Every task has 'path' property which I think is exactly what we need here :)